app/src/main/java/com/mk/tasky/agenda/presentation/home/AgendaItem.kt

```diff
@@ -0,0 +1,13 @@
+package com.mk.tasky.agenda.presentation.home
+
+import java.time.LocalDateTime
+
+data class AgendaItem(
+    val id: String,
+    val title: String,
+    val isDone: Boolean,
+    val description: String,
+    val type: HomeAgendaType,
+    val firstDatetime: LocalDateTime,
+    val secondDatetime: LocalDateTime? = null
```

Comment:
Hmm, I'm not sure about if it's a good idea to model it that way. This could make it a pain to extend in case there are more `HomeAgendaType`'s with individual behavior (such as `Event` which will require individual things like attendees and photos). Also, only tasks can be marked as done while your way of modeling it will add such a boolean to all types.

Why not model it somehow like this (this is from my project)? If you want to use the same name for the fields (like `id` instead of `eventId`), you'd need to make the child classes normal ones instead of data classes (but losing the advantages of data classes, which is why I'm fine with using different names)
```
sealed class AgendaItem(
    val id: String,
    val title: String,
    val description: String?,
    val sortDate: ZonedDateTime,
    val remindAt: ZonedDateTime,
    val reminderType: ReminderType
) {
    data class Event(
        val eventId: String,
        val eventTitle: String,
        val eventDescription: String?,
        val from: ZonedDateTime,
        val to: ZonedDateTime,
        val photos: List<EventPhoto>,
        val attendees: List<Attendee>,
        val isUserEventCreator: Boolean,
        val host: String?,
        val remindAtTime: ZonedDateTime,
        val eventReminderType: ReminderType
    ) : AgendaItem(eventId, eventTitle, eventDescription, from, remindAtTime, eventReminderType)

    data class Task(
        val taskId: String,
        val taskTitle: String,
        val taskDescription: String?,
        val time: ZonedDateTime,
        val isDone: Boolean,
        val remindAtTime: ZonedDateTime,
        val taskReminderType: ReminderType
    ) : AgendaItem(taskId, taskTitle, taskDescription, time, remindAtTime, taskReminderType)

    data class Reminder(
        val reminderId: String,
        val reminderTitle: String,
        val reminderDescription: String?,
        val time: ZonedDateTime,
        val remindAtTime: ZonedDateTime,
        val reminderReminderType: ReminderType
    ) : AgendaItem(reminderId, reminderTitle, reminderDescription, time, remindAtTime, reminderReminderType)
}
```

Reply:
Ohh that's interesting
Watchout cause the next commits are going to be big then since we are removing the Task and Reminder as we know it, and it's all going to inherit from AgendaItem